persistence/sql/identity/persister_identity.go

```diff
@@ -645,32 +645,37 @@ func (p *IdentityPersister) CreateIdentities(ctx context.Context, identities ...
 		if len(failedIdentityIDs) > 0 {
 			partialErr := &identity.CreateIdentitiesError{}
 			failedIDs := make([]uuid.UUID, 0, len(failedIdentityIDs))
+			succeededIDs := make([]uuid.UUID, 0, len(identities))
+
 			for _, ident := range identities {
 				if _, ok := failedIdentityIDs[ident.ID]; ok {
 					partialErr.AddFailedIdentity(ident, sqlcon.ErrUniqueViolation)
 					failedIDs = append(failedIDs, ident.ID)
+				} else {
+					succeededIDs = append(succeededIDs, ident.ID)
 				}
 			}
 			// Manually roll back by deleting the identities that were inserted before the
 			// error occurred.
 			if err := p.DeleteIdentities(ctx, failedIDs); err != nil {
 				return sqlcon.HandleError(err)
 			}
-			// Wrap the partial error with the first error that occurred, so that the caller
-			// can continue to handle the error either as a partial error or a full error.
+
+			// Partial success: report succeeded identities as created.
+			for _, identID := range succeededIDs {
+				span.AddEvent(events.NewIdentityCreated(ctx, identID))
```

Comment:
I don't think that we should emit the events in the transaction, as it might be retried?

Reply:
Good question - will a retry in the transaction cause this event to be duplicated?

Reply:
I think so, it will trace the event two times, which triggers to sends. So I agree, this should be outside of the transaction (like it was before)

Reply:
Yes, this must be outside the transaction block. Could use a `defer`.

Reply:
makes sense, thanks!